specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,322 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+
+@versioned(AzureCommunicationMessagesService.Versions)
+namespace AzureCommunicationMessagesService;
+
+@access(Access.public)
+@doc("The type of message. Supports text, image, template.")
+enum CommunicationMessageType {
+  @doc("Text message type.")
+  text,
+
+  @doc("Image message type.")
+  image,
+
+  @doc("Template message type.")
+  template,
+}
+
+@access(Access.public)
+@doc("The WhatsApp button sub type.")
+enum WhatsAppMessageButtonSubType {
+  @doc("The WhatsApp button sub type is quick reply.")
+  quickReply,
+
+  @doc("The WhatsApp button sub type is url.")
+  url,
+}
+
+@access(Access.public)
+@doc("The aggregated template status.")
+enum MessageTemplateStatus {
+  @doc("Message template is approved.")
+  approved,
+
+  @doc("Message template is rejected.")
+  rejected,
+
+  @doc("Message template is pending.")
+  pending,
+
+  @doc("Message template is paused.")
+  paused,
+}
+
+@discriminator("kind")
+@doc("Details of the message to send.")
+model NotificationContent {
+  @doc("The Channel Registration ID for the Business Identifier.")
+  channelRegistrationId: uuid;
+
+  @doc("The native external platform user identifiers of the recipient.")
+  to: string[];
+}
+
+@doc("A request to send a text notification.")
+model TextNotificationContent extends NotificationContent {
+  @doc("Message notification type is text.")
+  kind: "text";
```

Comment:
Should we extract this literal value out to a new `enum NotificationContentType`?

Reply:
extracted to enum CommunicationMessageType